src/mlir/generator.cc

```diff
@@ -130,6 +130,14 @@ namespace mlir::verona
           return std::move(err);
         // Associate function with module (late mangling)
         func->setAttr("class", TypeAttr::get(type));
+        // Add qualifiers as attributes
+        llvm::SmallVector<::ast::WeakAst, 4> quals;
+        AST::getFunctionQualifiers(quals, node);
+        llvm::SmallVector<mlir::Attribute, 4> qualAttrs;
+        for (auto qual : quals)
+          qualAttrs.push_back(
+            StringAttr::get(AST::getTokenValue(qual), context));
+        func->setAttr("qualifiers", ArrayAttr::get(qualAttrs, context));
```

Comment:
Shouldn't this be set only if the set of qualifiers is non-empty?

Reply:
Good idea, will do.